### PR TITLE
test: add tests for buffer.ts

### LINKED ISSUE #2995: Suggestion for Improving Test Coverage of utils/buffer.ts
### What is the feature you are proposing?

The equals function in utils/buffer.ts does not appear to be used.

https://app.codecov.io/github/honojs/hono/blob/main/src%2Futils%2Fbuffer.ts#L9

If it is a function used on the user side, I would suggest adding test code.
Or, if it is not going to be used in the future, it would be better to remove it.

If either of these fixes are needed, I will send out a pull request!
If no modifications are needed, the issue should be closed!

### PR DESCRIPTION
**Pull request by yasuaki640**
close #2995

Add some tests for uncovered lines.

### The author should do the following, if applicable

- [x] Add tests
- [x] Run tests
- [x] `bun run format:fix && bun run lint:fix` to format the code
- [ ] Add [TSDoc](https://tsdoc.org/)/[JSDoc](https://jsdoc.app/about-getting-started) to document the code

### CODE REVIEW

File: src/utils/buffer.test.ts [hunk 1]
```diff
@@ -43,6 +70,7 @@ describe('buffer', () => {
     expect(await timingSafeEqual({ a: 1 }, { a: 2 })).toBe(false)
     expect(await timingSafeEqual([1, 2], [1, 2])).toBe(false)
     expect(await timingSafeEqual([1, 2], [1, 2, 3])).toBe(false)
+    expect(await timingSafeEqual('a', 'b', () => undefined)).toBe(false)
```
Comment (yasuaki640) [inline src/utils/buffer.test.ts:73]: Test covering the first if statement branch

https://github.com/honojs/hono/blob/7ee829be6bef89746a0157c007ea9668141ddce9/src/utils/buffer.ts#L41-L43

File: src/utils/buffer.test.ts [hunk 2]
```diff
@@ -1,5 +1,32 @@
 import { SHA256 as sha256CryptoJS } from 'crypto-js'
-import { bufferToFormData, bufferToString, timingSafeEqual } from './buffer'
+import { bufferToFormData, bufferToString, equal, timingSafeEqual } from './buffer'
+import { expect } from 'vitest'
```
Comment (yusukebe) [inline src/utils/buffer.test.ts:3]: Is this line necessary? I think the `expect` function is already imported globally.

Comment (yasuaki640) [reply src/utils/buffer.test.ts]: Okay. I've deleted this import.

**Review by yusukebe — APPROVED** [review]
LGTM!

**Comment by yusukebe** [comment]
@yasuaki640 Thanks!

**Pull request merged**